app/controllers/course/users_controller.rb

```diff
@@ -0,0 +1,52 @@
+class Course::UsersController < Course::ModuleController
+  load_and_authorize_resource
+  before_action :ensure_not_registered, only: [:new, :create]
+
+  def index # :nodoc:
+  end
+
+  def new # :nodoc:
+  end
+
+  def create # :nodoc:
+    if @registration.save
+      success = t('.success', role: t("course.users.role.#{@registration.role}"))
+      redirect_to course_path(@course), flash: { success: success }
+    else
+      render 'new'
+    end
+  end
+
+  def update # :nodoc:
+    @registration.update(registration_params)
+    if @registration.save
+      success = t('.success', role: t("course.users.role.#{@registration.role}"))
+      redirect_to course_path(@course), flash: { success: success }
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy # :nodoc:
+    if @registration.destroy
+      success = t('.success', role: @registration.role, email: @registration.user.email)
+      redirect_to course_users_path(@course, flash: { success: success })
+    else
+      render 'edit'
+    end
+  end
+
+  private
+
+  def registration_params # :nodoc:
+    params.require(:course_user).permit(:role, :name)
+  end
+
+  def ensure_not_registered # :nodoc:
+    if current_course_user
```

Comment:
Use a guard clause instead of wrapping the code inside a conditional expression.